### ISSUE TITLE
Need help: JS won't generate an output (compiled, but not written.)

### ISSUE THREAD

**Issue by philer** [brunch/brunch#1248]
My code is structured in es6 modules with a single entry point. I want to generate an output file that contains all other modules that are imported in the file. I'm using babel. I'm also using less, which works fine. There is no magic with external modules or vendor directories, just my own code. Here's my setup:

```
js/
├── index.js
├── otherModules.js
└── ...
less/
├── index.less
├── otherStyles.less
└── ...
dist/
package.json
rollup.config.js
index.html
```
I want to have my output in `dist/`, one js and one css file. Note, that `index.html` needs to stay where it is.

The problem is that there is no output written for my javascript. However css works fine:
```
$ build b
22 Mar 01:56:01 - warn: js/index.js compiled, but not written. Check your javascripts.joinTo config
22 Mar 01:56:01 - info: compiled index.less into brunched.css in 1 sec
```
After running the build my `dist/` only contains the generated `brunched.css` and its map file. That file does include all the other styles that are imported within `index.less`, so why does the same not work for js?

dependencies in `package.json`:
```json
  "devDependencies": {
    "babel-brunch": "^6.0.2",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-stage-2": "^6.5.0",
    "brunch": "^2.5.1",
    "javascript-brunch": "^2.0.0",
    "less-brunch": "^2.0.0",
    "less-plugin-clean-css": "^1.5.1",
    "uglify-js-brunch": "^2.0.1"
  },
```
`brunch-config.coffee`
```coffee
module.exports = config:
  paths:
    public: 'dist'
    watched: ['js', 'less']
  files:
    javascript:
      joinTo:
        'brunched.js': 'js/index.js'
    stylesheets:
      joinTo:
        'brunched.css': 'less/index.less'
  conventions:
    ignored: (str) -> (str != 'js/index.js' && str != 'less/index.less')
  plugins:
    babel:
      presets: ['es2015', 'stage-2']
  # modules:
  #   wrapper: (_, data) -> '(function(global, document) {' + data + '})(window, document);'
  #   definition: false
```
You can see that I tried to define my own wrapper (a simple iife). I'm writing a standalone app, so I don't really want any unnecessary commonJS wrapping or what have you. However that did nothing to change the result.

### Environment

1. Brunch: 2.5.1
2. Node: 5.9.0
3. NPM: 3.7.3
4. Operating system: Linux Mint 17.1 (essentially Ubuntu 14.04)

**Comment by philer** [comment]
Nevermind I was too stupid to put an 's' at the end of 'javascripts' in my config.

**Comment by goshakkk** [comment]
Cool :+1:

**Comment by philer** [comment]
Heh @goshakkk 
On a side note the whole single entry point thing still doesn't work. I'm guessing brunch isn't designed to automatically follow imports like rollup or webpack to automatically create a minimal package. At least that's what I got from [here](https://github.com/brunch/brunch-guide/issues/28).

**Comment by goshakkk** [comment]
Brunch does follow the imports, but currently only when it comes to node_modules. App code, all of it is included, though there is not much stopping us from having an option to specify an entry point and just follow its imports in the future

cc @paulmillr

**Comment by razor-x** [comment]
@philer Just tried to use  [istr/rollup-brunch](https://github.com/istr/rollup-brunch) as well and ran into the same issue: it didn't actually do anything. I also tried the fork here [judas-christ/rollup-brunch](https://github.com/judas-christ/rollup-brunch).

I did get it "working" but it's not ideal. Here is a minimal working demo: https://github.com/rxbugs/brunch-rollup-test

Probably not too hard to get this plugin up to snuff though.

**TLDR;** Use the fork and don't use babel-brunch, inject the rollup babel plugin per the README, set `config.plugins.javascript.validate = false`.

Here is a summary of what I found.

1. The fork is a bit more agnostic as it doesn't assume any rollup plugins: you need to inject them in from the brunch config (similar to how postcss-brunch works). I like this since it's not "rollup-babel-brunch".

2. If you try and use it with babel-brunch, the output seems to ignore rollup (no tree shaking, etc.) Not sure why this is as I'm not familiar with plugin priority, but I'm guessing babel-brunch is processing it normally, and rollup-brunch never gets the files?

3. If you use it without babel-brunch you will get errors from javascript-brunch which you can disable by setting `config.plugins.javascript.validate = false`. I'm guessing this happens becasue javascript-brunch is getting the js before it is processed by rollup (so the validation fails on `import`, `export`). Ideally rollup-brunch would handle things like babel-brunch does and avoid this issue.

4. Using this with auto-reload-brunch triggers an `eval` warning from rollup.

5. I didn't test this with yet with `entryPoints`.
